src/connector/src/parser/protobuf/parser.rs

```diff
@@ -250,30 +381,97 @@ pub fn from_protobuf_value(field_desc: &FieldDescriptor, value: &Value) -> Resul
             ScalarImpl::Utf8(enum_symbol.name().into())
         }
         Value::Message(dyn_msg) => {
-            let mut rw_values = Vec::with_capacity(dyn_msg.descriptor().fields().len());
-            // fields is a btree map in descriptor
-            // so it's order is the same as datatype
-            for field_desc in dyn_msg.descriptor().fields() {
-                // missing field
-                if !dyn_msg.has_field(&field_desc)
-                    && field_desc.cardinality() == Cardinality::Required
-                {
-                    let err_msg = format!(
-                        "protobuf parse error.missing required field {:?}",
-                        field_desc
-                    );
-                    return Err(RwError::from(ProtocolError(err_msg)));
+            let any_flag;
+            if let Some(&DataType::Jsonb) = type_expected {
+                any_flag = true;
+            } else {
+                any_flag = false;
+            }
```

Comment:
This can be more precise to decide whether the protobuf value is `Any`.
```rust
if let Kind::Message(msg_desc) = field_desc.kind() {
    any_flag = msg_desc.full_name() == "google.protobuf.Any";
}
```

Reply:
We indeed have the full name in the field descriptor inside dynamic message. 🤣
```rust
let any_flag = dyn_msg.descriptor().full_name() == "google.protobuf.Any";

if any_flag {
    debug_assert!(
        type_expected == Some(&DataType::Jsonb),
        "`type_expected` must be of `DataType::Jsonb` for any protobuf type"
    );
}
```